docs/extend/extend-cli/cli-developing-a-plugin.md

```diff
@@ -1,219 +1,348 @@
-# Developing a new plug-in
+# Developing a new Zowe CLI plug-in
 Before you begin this tutorial, complete the [Extending an existing plug-in](cli-extending-a-plugin.md) tutorial.
 
 ## Overview
-This tutorial demonstrates how to create a brand new Zowe&trade; CLI plug-in that uses Zowe CLI Node.js programmatic APIs.
+The advantage of Zowe CLI and of the CLI approach in mainframe development is that it allows for combining different developer tools for new and interesting uses.
+
+This tutorial demonstrates how to create a brand new Zowe CLI plug-in that uses Node.js to create a client-side API.
+
+After following all the steps, you will have created a data set diff utility plug-in called **Files Util Plug-in**. This plug-in takes in any two data sets, or files, and returns a plain text output in the terminal showing how they differ. This tutorial will also show you how you can integrate your new plug-in with a third-party utility to make your output colorful and easier to read, as shown in the image at the [bottom of this page](../extend-cli/cli-developing-a-plugin#bringing-together-new-tools).
+
+If you are ready to create your own unique Zowe CLI plug-in, refer to the notes at the end of each tutorial step for guidance.
+
+If you are interested in creating a credential manager plug-in, see the [Zowe CLI secrets for kubernetes plug-in](https://github.com/zowe/zowe-cli-secrets-for-kubernetes) repository.
 
-At the end of this tutorial, you will have created a data set diff utility plug-in for Zowe CLI, from which you can pipe
-your plugin's output to a third-party utility for a side-by-side diff of data set member contents.
+## Setting up the new sample plug-in project
+ Download the sample plug-in source and delete the irrelevant content to set up your plug-in project.
```

Comment:
I love these overviews. Wondering if you think some text styling to somewhat separate out the overviews from the rest of the text is doable/helpful

Reply:
Don't really have a lot of formatting/styling options for these. Will have to leave these as they are.